drivers/storage/portworx/component/monitoring.go

```diff
@@ -159,6 +221,185 @@ func (c *monitoring) createPrometheusRule(
+	return volumes, volumeMounts
+}
+
+func (c *monitoring) getGrafanaDeploymentSpec(
+	cluster *corev1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+	imageName string,
+	volumes []v1.Volume,
+	volumeMounts []v1.VolumeMount,
+	cpuQuantity resource.Quantity,
+	memoryQuantity resource.Quantity,
+) *appsv1.Deployment {
+	deploymentLabels := map[string]string{
+		"app": "grafana",
+	}
+	templateLabels := map[string]string{
+		"app": "grafana",
+	}
+
+	replicas := int32(1)
+	imagePullPolicy := pxutil.ImagePullPolicy(cluster)
+
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            GrafanaDeploymentName,
+			Namespace:       cluster.Namespace,
+			Labels:          deploymentLabels,
+			OwnerReferences: []metav1.OwnerReference{*ownerRef},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: templateLabels,
+			},
+			Replicas: &replicas,
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: templateLabels,
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:            GrafanaContainerName,
+							Image:           imageName,
+							ImagePullPolicy: imagePullPolicy,
+							Resources: v1.ResourceRequirements{
+								Requests: map[v1.ResourceName]resource.Quantity{
+									v1.ResourceCPU: cpuQuantity,
+								},
+								Limits: map[v1.ResourceName]resource.Quantity{
+									v1.ResourceMemory: memoryQuantity,
+								},
+							},
+							ReadinessProbe: &v1.Probe{
+								ProbeHandler: v1.ProbeHandler{
+									HTTPGet: &v1.HTTPGetAction{
+										Path: grafanaLoginEndpoint,
+										Port: intstr.FromInt(grafanaPort),
+									},
+								},
+							},
+							VolumeMounts: volumeMounts,
+						},
+					},
+					Volumes: volumes,
+				},
+			},
+		},
+	}
+
+	if cluster.Spec.ImagePullSecret != nil && *cluster.Spec.ImagePullSecret != "" {
+		deployment.Spec.Template.Spec.ImagePullSecrets = append(
+			[]v1.LocalObjectReference{},
+			v1.LocalObjectReference{
+				Name: *cluster.Spec.ImagePullSecret,
+			},
+		)
+	}
+
+	if cluster.Spec.Placement != nil {
+		if cluster.Spec.Placement.NodeAffinity != nil {
+			deployment.Spec.Template.Spec.Affinity.NodeAffinity =
+				cluster.Spec.Placement.NodeAffinity.DeepCopy()
+		}
+
+		if len(cluster.Spec.Placement.Tolerations) > 0 {
+			deployment.Spec.Template.Spec.Tolerations = make([]v1.Toleration, 0)
+			for _, toleration := range cluster.Spec.Placement.Tolerations {
+				deployment.Spec.Template.Spec.Tolerations = append(
+					deployment.Spec.Template.Spec.Tolerations,
+					*(toleration.DeepCopy()),
+				)
+			}
+		}
+	}
+
+	// If resources is specified in the spec, the resources specified by annotation (such as portworx.io/autopilot-cpu)
+	// will be overwritten.
+	if cluster.Spec.Autopilot.Resources != nil {
+		deployment.Spec.Template.Spec.Containers[0].Resources = *cluster.Spec.Autopilot.Resources
+	}
```

Comment:
Let's remove this as it does not apply here.

Reply:
Removed